### PR TITLE
Fixes #28610 - Remove dropped API endpoints from hammer

### PR DESCRIPTION
**Pull request by shiramax**

### CODE REVIEW

File: lib/hammer_cli_foreman/associating_commands.rb [hunk 1]
```diff
@@ -229,23 +229,23 @@ class RemoveUsergroupCommand < HammerCLIForeman::RemoveAssociatedCommand
...
+      class AddProvisioningTemplateCommand < HammerCLIForeman::AddAssociatedCommand
+        associated_resource :provisioning_templates
+        desc _("Associate a Provisioning template")
 
         success_message _("The configuration template has been associated.")
-        failure_message _("Could not associate the configuration template")
+        failure_message _("Could not associate the Provisioning template")
       end
 
       class RemoveConfigTemplateCommand < HammerCLIForeman::RemoveAssociatedCommand
```
Comment (tbrisker) [inline lib/hammer_cli_foreman/associating_commands.rb]: Should this also be renamed?
```suggestion
       class RemoveProvisioningTemplateCommand < HammerCLIForeman::RemoveAssociatedCommand
```

File: lib/hammer_cli_foreman/associating_commands.rb [hunk 2]
```diff
@@ -229,23 +229,23 @@ class RemoveUsergroupCommand < HammerCLIForeman::RemoveAssociatedCommand
...
-    module ConfigTemplate
+    module ProvisioningTemplate
       extend CommandExtension
 
-      class AddConfigTemplateCommand < HammerCLIForeman::AddAssociatedCommand
-        associated_resource :config_templates
-        desc _("Associate a configuration template")
+      class AddProvisioningTemplateCommand < HammerCLIForeman::AddAssociatedCommand
+        associated_resource :provisioning_templates
+        desc _("Associate a Provisioning template")
```
Comment (tbrisker) [inline lib/hammer_cli_foreman/associating_commands.rb]: I don't think this needs capitalization:
```suggestion
        desc _("Associate a provisioning template")
```

File: lib/hammer_cli_foreman/associating_commands.rb [hunk 3]
```diff
@@ -229,23 +229,23 @@ class RemoveUsergroupCommand < HammerCLIForeman::RemoveAssociatedCommand
...
 
         success_message _("The configuration template has been associated.")
-        failure_message _("Could not associate the configuration template")
+        failure_message _("Could not associate the Provisioning template")
```
Comment (tbrisker) [inline lib/hammer_cli_foreman/associating_commands.rb]: here too:
```suggestion
        failure_message _("Could not associate the provisioning template")
```

File: lib/hammer_cli_foreman/associating_commands.rb [hunk 4]
```diff
@@ -229,23 +229,23 @@ class RemoveUsergroupCommand < HammerCLIForeman::RemoveAssociatedCommand
...
       extend CommandExtension
 
-      class AddConfigTemplateCommand < HammerCLIForeman::AddAssociatedCommand
-        associated_resource :config_templates
-        desc _("Associate a configuration template")
+      class AddProvisioningTemplateCommand < HammerCLIForeman::AddAssociatedCommand
+        associated_resource :provisioning_templates
+        desc _("Associate a Provisioning template")
 
         success_message _("The configuration template has been associated.")
```
Comment (xprazak2) [inline lib/hammer_cli_foreman/associating_commands.rb]: s/configuration/provisioining/

File: lib/hammer_cli_foreman/associating_commands.rb (hunk 1 above)
Comment (xprazak2) [inline lib/hammer_cli_foreman/associating_commands.rb]: Should we rename this to `RemoveProvisioningTemplateCommand`?

File: lib/hammer_cli_foreman/associating_commands.rb [hunk 5]
```diff
@@ -229,23 +229,23 @@ class RemoveUsergroupCommand < HammerCLIForeman::RemoveAssociatedCommand
...
-        failure_message _("Could not associate the configuration template")
+        failure_message _("Could not associate the Provisioning template")
       end
 
       class RemoveConfigTemplateCommand < HammerCLIForeman::RemoveAssociatedCommand
-        associated_resource :config_templates
+        associated_resource :provisioning_templates
         desc _("Disassociate a configuration template")
```
Comment (xprazak2) [inline lib/hammer_cli_foreman/associating_commands.rb]: s/configuration/provisioining/

**Comment by xprazak2** [comment]
`hammer template combination list` no longer recognizes `--provisioning-template-id` option.

I think we should also update the `template` command description in the output of `hammer --help`

File: lib/hammer_cli_foreman/associating_commands.rb (hunk 4 above)
Comment (rahulbajaj0509) [reply lib/hammer_cli_foreman/associating_commands.rb]: good catch!

**Comment by shiramax** [comment]
@tbrisker @xprazak2  @rahulbajaj0509  I fixed everything, thanks!

File: lib/hammer_cli_foreman/associating_commands.rb [hunk 6]
```diff
@@ -229,23 +229,23 @@ class RemoveUsergroupCommand < HammerCLIForeman::RemoveAssociatedCommand
...
-        success_message _("The configuration template has been associated.")
-        failure_message _("Could not associate the configuration template")
+        success_message _("The provisioning template has been associated.")
+        failure_message _("Could not associate the provisioning template")
       end
 
-      class RemoveConfigTemplateCommand < HammerCLIForeman::RemoveAssociatedCommand
-        associated_resource :config_templates
-        desc _("Disassociate a configuration template")
+       class RemoveProvisioningTemplateCommand < HammerCLIForeman::RemoveAssociatedCommand
```
Comment (tbrisker) [inline lib/hammer_cli_foreman/associating_commands.rb]: looks like indentation is off here?

File: lib/hammer_cli_foreman/combination.rb [hunk 7]
```diff
@@ -45,8 +45,8 @@ class ListCombination < HammerCLIForeman::ListCommand
       end
 
       build_options do |o|
-        o.expand(:all).except(:config_templates, :hostgroups, :environments)
-        o.without(:config_template_id, :hostgroup_id, :environment_id)
```
Comment (tbrisker) [inline lib/hammer_cli_foreman/combination.rb]: is the removal from the `o.without` here and below intentional?

Comment (shiramax) [reply lib/hammer_cli_foreman/combination.rb]: @tbrisker, yes @xprazak2 mentioned that the template_id should be in the command.

**Pull request merged**